From c558e217d0b80a50f7ea81c678f534907e736b99 Mon Sep 17 00:00:00 2001
From: Zhou Xuemei <zhouxuemei@wxiat.com>
Date: Tue, 9 Aug 2022 08:30:54 +0800
Subject: [PATCH] =?UTF-8?q?sw64:=20iommu:=20allow=C2=A0unlimited=C2=A0mini?=
 =?UTF-8?q?mum=C2=A0value=C2=A0of=C2=A0iova=C2=A0in=C2=A0unmanaged=C2=A0do?=
 =?UTF-8?q?main?=
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c558e217d0b80a50f7ea81c678f534907e736b99
Modified-by-SEL: No

MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

IO virtual address assignment in unmanaged domain is completely handed
over to the device owner, so there is no need to set the IOVA baseline.

Signed-off-by: Zhou Xuemei <zhouxuemei@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/iommu/sw64/sunway_iommu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/sw64/sunway_iommu.c b/drivers/iommu/sw64/sunway_iommu.c
index b6c8f1272d28..580619c6a571 100644
--- a/drivers/iommu/sw64/sunway_iommu.c
+++ b/drivers/iommu/sw64/sunway_iommu.c
@@ -1382,7 +1382,7 @@ static struct iommu_domain *sunway_iommu_domain_alloc(unsigned type)
 
 		sdomain->pt_root = (void *)get_zeroed_page(GFP_KERNEL);
 
-		sdomain->domain.geometry.aperture_start = SW64_DMA_START;
+		sdomain->domain.geometry.aperture_start = 0ULL;
 		sdomain->domain.geometry.aperture_end	= (~0ULL);
 		sdomain->domain.geometry.force_aperture	= true;
 		sdomain->type = IOMMU_DOMAIN_UNMANAGED;
-- 
2.33.0

